cumulus/ansible/tasks/playbooks/roles/ec2-pod/tasks/launch.yml

```diff
@@ -0,0 +1,157 @@
+---
+# Note that we don't include master/node ami variables - these
+# could be conditionally checked, but because the next thing we
+# do is try to find the master/node ami if they aren't defined
+# it seems unnecessary.
+- name: Fail early if required variables are not defined
+  fail:
+    msg: "Variable {{ item }} is not defined"
+  when: item not in hostvars[inventory_hostname]
+  with_items:
+    - aws_keyname
+    - master_instance_type
+    - node_instance_type
+    - node_instance_count
+    - cluster_id
+    - cluster_region
+    - cluster_zone
+
+
+- name: Define default firewall rules if not provided
+  set_fact:
+    ec2_pod_rules:
+      - proto: tcp # ssh
+        from_port: 22
+        to_port: 22
+        cidr_ip: "{{ ssh_cidr_ip }}"
+
+      - proto: all
+        group_name: "ec2_pod_{{cluster_id}}"
+        group_desc: "Ec2 security group for cluster {{ cluster_id }}"
+  when: ec2_pod_rules is not defined
+
+
+# Find the Master AMI if master_instance_ami is not specified
+- block:
+    - name: Find latest Master AMI
+      ec2_ami_find:
+        state: available
+        sort: creationDate
+        sort_order: descending
+        owner: "{{ master_ami_spec.owner }}"
+        ami_tags: "{{ master_ami_spec.tags }}"
+        region: "{{ cluster_region }}"
+      register: master_amis
+
+    - name: Register to master_instance_ami variable
+      set_fact:
+        master_instance_ami: "{{ master_amis.results[0]['ami_id'] }}"
+  when: master_instance_ami is not defined
+
+# Find the Node AMI if node_instance_ami is not specified
+- block:
+    - name: Find latest node AMI
+      ec2_ami_find:
+        state: available
+        sort: creationDaten
```

Comment:
We could modify the integration test to select an AMI using tags to catch this sort of error in the future.